R/bmerge.R

```diff
@@ -71,7 +71,13 @@ bmerge = function(i, x, icols, xcols, roll, rollends, nomatch, mult, ops, verbos
       stopf("Incompatible join types: %s (%s) and %s (%s). Factor columns must join to factor or character columns.", xname, xclass, iname, iclass)
     }
     if (xclass == iclass) {
-      if (verbose) catf("%s has same type (%s) as %s. No coercion needed.\n", iname, xclass, xname)
+      if (length(icols)>1 && is(x[[xc]], "Date") && is(i[[ic]], "Date")) {
```

Comment:
To modify only when strictly needed, could be added the next condition?
```r
&& identical(storage.mode(x[[xc]]), storage.mode(i[[ic]]))
```
(following https://bugs.r-project.org/show_bug.cgi?id=18782#c9)

Reply:
I would use `inherits()`, `is()` is for S4 classes IMO (even though it happens to work for S3 classes).

Reply:
BTW, IINM `POSIXct` can have the same concern -- {base} doesn't care about storing `POSIXct` as `integer` sometimes if possible.

Reply:
> I would use `inherits()`, `is()` is for S4 classes IMO (even though it happens to work for S3 classes).

Previously I checked if `class(x[[xc]])=="Date"` are equal and our linter nudged me towards `is`, so we might also want to adjust the linter :D

Reply:
`class(x)=="Date"` is bad because `class()` can be `length>1`, c.f. `class(Sys.time())`, hence `==` is rarely intended.

The linter mentions `is()` as well as `inherits()` to cover all bases since there's no `is.Date()` and we didn't hand-code anything to cover base S3 classes:

```r
lintr::lint("class(x) == 'Date'\n", lintr::class_equals_linter())
# <text>:1:1: warning: [class_equals_linter] Instead of comparing class(x) with ==, use inherits(x, 'class-name') or is.<class> or is(x, 'class').
# class(x) == 'Date'
# ^~~~~~~~~~~~~~~~~~
```

Reply:
My bad, seems that I overlooked the `inherits` part.